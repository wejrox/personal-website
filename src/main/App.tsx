import React from "react";
import FloatingExternalLinks from "./core/FloatingExternalLinks";
import NavigationBar from "./core/navigation/NavigationBar";
import Footer from "./core/Footer";

class App extends React.Component {
    public render() {
        return (
            <>
                <FloatingExternalLinks/>
                <NavigationBar/>
                <Footer/>
            </>
        );
    }
}

export default App;
