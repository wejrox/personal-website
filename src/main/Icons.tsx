import * as React from "react";
import {GoMarkGithub} from "react-icons/go";

export enum Icon {
    GoMarkGithub
}

class Icons {
    public static RenderIcon(icon: Icon): JSX.Element {
        switch (icon) {
            case Icon.GoMarkGithub:
                return <GoMarkGithub/>;
            default:
                console.error("Icon requested is not supported.");
                return <></>;
        }
    }
}

export default Icons;