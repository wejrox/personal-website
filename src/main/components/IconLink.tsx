import * as React from "react";
import Icons, {Icon} from "../Icons";

interface IconLinkProps {
    link: string;
    icon: Icon;
}

class IconLink extends React.Component<IconLinkProps> {

    public render() {
        return (
            <div>
                <a href={this.props.link}>
                    {Icons.RenderIcon(this.props.icon)}
                </a>
            </div>
        );
    }
}

export default IconLink;