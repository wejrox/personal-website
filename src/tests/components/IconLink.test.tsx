import * as Enzyme from "enzyme";
import * as React from "react";
import IconLink from "../../main/components/IconLink";
import { Icon } from "../../main/Icons";

describe("IconLink Component", () => {
    const link: string = "https://www.google.com.au";
    const icon: Icon = Icon.GoMarkGithub;
    const wrapper = Enzyme.shallow(<IconLink link={link} icon={icon}/>);

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});