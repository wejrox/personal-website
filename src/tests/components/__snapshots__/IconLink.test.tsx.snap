// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink Component renders correctly 1`] = `ShallowWrapper {}`;
